core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2388,15 +2388,17 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
 
     val g = (b: B, a: A) => f(a, b)
 
-    ZIO.transplant { graft =>
-      ZIO.descriptorWith { d =>
-        new ZIO.RaceWith(
-          graft(self),
-          graft(that),
-          coordinate(d.id, f, true),
-          coordinate(d.id, g, false),
-          trace
-        )
+    ZIO.uninterruptibleMask { restore =>
+      ZIO.transplant { graft =>
+        ZIO.descriptorWith { d =>
```

Comment:
`ZIO.fiberId.flatMap(...)` (or `ZIO.fiberIdWith,` if it existed).

Reply:
I think we were trying to avoid the extra `flatMap` but I added `fiberIdWith` so we can use that now.